core/src/main/java/org/mule/runtime/core/util/IOUtils.java

```diff
@@ -262,4 +272,60 @@ public static long copyLarge(Reader input, Writer output) throws IOException
         }
         return count;
     }
+
+    /**
+     * Transforms an Object attachment into a DataHandler of its corresponding type.
+     *
+     * @param name        the name of the attachment being handled
+     * @param object      the attachment to be handled
+     * @param contentType the Content-Type of the attachment that is being handled
+     * @return a {@link DataHandler} of the corresponding attachment
+     * @throws Exception if the transformation fails.
+     */
+    public static DataHandler toDataHandler(String name, Object object, String contentType) throws Exception
```

Comment:
This method should be replaced for a common one that @afelisatti is using for the HTTP extension.